Libraries/Opc.Ua.Configuration/ApplicationInstance.cs

```diff
@@ -821,19 +888,72 @@ public void OnCertificateValidation(object sender, CertificateValidationEventArg
                 Utils.GetAbsoluteDirectoryPath(id.StorePath, true, true, true);
             }
 
+            var builder = CertificateFactory.CreateCertificate(
```

Comment:
I would ike the CertificateFactory.CreateCertificate to have a method 
I provide the certificateType and it creates the RSA or ECC Certificate with the right curve for me depending on the provided certificate type, or else everywhere I create a certificate I need to check the type manually and create the RSA/ECC cert depending on the type. I would also like a global List  Class/Enum of Certificate Types (maybe like the well known roles) because in the code ((especially gds) i see a lot of passing around certificate type as string / Node Id and then matching it to a Dictionary of Object Type Ids which i not really see as good practice.